src/anyp/Uri.cc

```diff
@@ -167,6 +167,20 @@ urlParseProtocol(const char *b)
     return AnyP::PROTO_NONE;
 }
 
+bool
+urlAppendDomain(char *host)
+{
+    if (Config.appendDomain && !strchr(host, '.') && !strchr(host, ':')) {
+        const auto dlen = strlen(host);
+        if (dlen > (SQUIDHOSTNAMELEN - Config.appendDomainLen - 1)) {
+            debugs(23, 2, "URL domain too large (" << dlen << " bytes)");
+            return false;
+        }
+        strncat(host, Config.appendDomain, SQUIDHOSTNAMELEN - dlen - 1);
```

Comment:
Either we trust our math or we do not:

```suggestion
        strncat(host, Config.appendDomain, Config.appendDomainLen);
```

I do not insist on this change.

Reply:
We used that way back. But static analysis showed that the terminator can be elided if the to-be-copied amount does not account for it. In this case that would result in random heap or stack memory being copied into the URL strings between the appended portion and the buffer end terminator (assuming we remember to put that in).

So current strncat() practice is to indicate the available buffer space and let the function put the terminator in the right place for us if it happens early. Yes we still need the -1 hack for end-of-buffer termination case, but that is better than mixing in sprinkles of +1 hacks into the math too.

Reply:
I do not insist on this change, so I am resolving this request to save time on discussing problems with your facts/arguments.